scipy/stats/_distribution_infrastructure.py

```diff
@@ -5219,3 +3582,288 @@ def __rtruediv__(self, other):
 
     def __neg__(self):
         return self * -1
+
+
+class Mixture(_ProbabilityDistribution):
+    r""" Class that represents a mixture distribution.
+
+    A mixture distribution is the distribution of a random variable
+    defined in the following way: first, a random variable is selected
+    from `components` according to the probabilities given by `weights`, then
+    the random variable is realized[1]_.
+
+    Parameters
+    ----------
+    components : sequence of `ContinuousDistribution`
+        The underlying instances of `ContinuousDistribution`.
+        All must have scalar shape parameters.
+    weights : sequence of floats
+        The corresponding probabilities of selecting each random variable.
+        Must be non-negative and sum to one.
+
+    Methods
+    -------
+    support
+
+    sample
+
+    moment
+
+    mean
+    median
+    mode
+
+    variance
+    standard_deviation
+
+    skewness
+    kurtosis
+
+    pdf
+    logpdf
+
+    cdf
+    icdf
+    ccdf
+    iccdf
+
+    logcdf
+    ilogcdf
+    logccdf
+    ilogccdf
+
+    entropy
+
+    Notes
+    -----
+    The following abbreviations are used throughout the documentation.
+
+    - PDF: probability density function
+    - CDF: cumulative distribution function
+    - CCDF: complementary CDF
+    - entropy: differential entropy
+    - log-*F*: logarithm of *F* (e.g. log-CDF)
+    - inverse *F*: inverse function of *F* (e.g. inverse CDF)
+
+    References
+    ----------
+    .. [1] Mixture distribution, *Wikipedia*,
+           https://en.wikipedia.org/wiki/Mixture_distribution
+
+    """
+    # Todo:
+    # Fix Normal(mu=0.5).logentropy() runtime warning
+    # Add support for array shapes, weights
+
+    def _input_validation(self, components, weights):
+        if len(components) == 0:
+            message = ("`components` must contain at least one random variable.")
+            raise ValueError(message)
+
+        for var in components:
+            # will generalize to other kinds of distributoins when there
+            # *are* other kinds of distributions
+            if not isinstance(var, ContinuousDistribution):
+                message = ("Each element of `components` must be an instance of "
+                           "`ContinuousDistribution`.")
+                raise ValueError(message)
+            if not var._shape == ():
+                message = "All elements of `components` must have scalar shapes."
+                raise ValueError(message)
+
+        if weights is None:
+            return
+
+        weights = np.asarray(weights)
+        if len(components) != len(weights):
```

Comment:
Maybe check weights.shape == (len,) to refuse inputs like len == 6 and weights.shape == (2, 3).

Reply:
Yeah, I think I had in mind to do:
```suggestion
        if weights.shape != (len(components),)
```
That's part of why I made `weights` an array. But I suppose I got distracted.

Reply:
You’re right, I thought `len` was `size`. But I’d propose exactly what you wrote: `if weights.shape != (len(components),)`